src/zino/state.py

```diff
@@ -42,8 +43,11 @@ class ZinoState(BaseModel):
     def dump_state_to_file(self, filename: str):
         """Dumps the full state to a file in JSON format"""
         _log.debug("dumping state to %s", filename)
-        with open(filename, "w") as statefile:
-            statefile.write(self.model_dump_json(exclude_none=True, indent=2))
+        copied_state = self.model_copy()
+        temp_file = "temporary-state.json"
```

Comment:
I think this sends a better signal as to the content/purpose of this file for anyone "accidentally" seeing it in the file system:

```suggestion
        temp_file = f"{filename}.tmp"
```